### ISSUE TITLE
Summer Sessions Suggestions!

### ISSUE THREAD

**Issue by katieannemills** [minisciencegirl/studyGroup#107]
**SUPER IMPORTANT EDIT**: Fill out this poll by May 6 to vote for the time you'd like to meet for summer session (May-August): http://whenisgood.net/zqr53wz

Hi folks,

We had an awesome lineup of sessions the past four months - but now that winter semester is wrapping up, it's time to wrangle a new set of lessons for the summer! So - what do people want to teach? What do people want to learn? Post suggestions here!

**Comment by wmhall** [comment]
I'd like to put together an R lesson on working with nested data frames using tidyr and purrr.

**Comment by katieannemills** [comment]
@wmhall awesome, let's do it!

@radaniba still interested in machine learning ala #101? (I certainly am!)

@minisciencegirl did you get the contact deets for the raffle winner at #ubcdsd?

Hey that's ~3 already, we only need like 8 or 9 to get us to September - who else has one?

**Comment by radaniba** [comment]
@BillMills totally

**Comment by radaniba** [comment]
@BillMills the plan in #101 will probably require 4 or 5 rounds at least how are you planning to schedule this ?

**Comment by markjohnsonubc** [comment]
"@minisciencegirl did you get the contact deets for the raffle winner at #ubcdsd?"
that'd be me. I agreed to lead a session on sonifying time series data

**Comment by SimonGoring** [comment]
I'd love to see something about Jekyll, Sphinx or something like that, building dynamic webpages.  Getting a bit away from R, but I think it would be helpful.

**Comment by jrebane** [comment]
Is there an appetite for any of the following?
- Natural language processing
NLTK in Python/tm in R
-

**Comment by jrebane** [comment]
Is there an appetite for any of the following?
- Natural language processing
  - NLTK in Python/tm in R
  - Sentiment analysis
  - Markov chain modeling (Probablistic language modeling)
- Online Data Visualization
  - Shiny (R)
  - The wide world of D3.js
- Network Analysis (R/Python)
  - Social network analysis
  - Visualization (graphviz and DiagrammeR)
  - Dynamic modeling of disease (More Epidemiology-ish, but maybe cool!)

Also, as a learner, I would love it if anyone could teach a neural networks/deep learning session at any point!

**Comment by wmhall** [comment]
@jrebane These all sound great. I'd be very interested in seeing the R side of natural language processing and network analysis.

**Comment by katieannemills** [comment]
@markjohnsonubc yes! that talk sounds awesome, I am v. looking forward to it - we're also going to re-vote the time for our meetings soon, so when we have a slot you are welcome to first pick of which one is more convenient for you.

@jrebane these are all awesome, and sound like they'll have some overlap with @radaniba's machine learning sessions - should we possibly coordinate there?

@radaniba that's up to what works for you - we could totally run all 4 or 5 sessions, possibly at the expense of some of the coworking days. Or, we could do this as a web series in parallel with the regular sessions, since I know it's pretty inconvenient for you to make it up to UBC. Would be great to think about packaging this up as a little mini-series for sharing and reuse, too!

**Comment by radaniba** [comment]
@BillMills web series like live broadcast ? or recorded sessions

**Comment by katieannemills** [comment]
@radaniba I was thinking both! But totally up to you if you're the one in front of the camera :)

**Comment by radaniba** [comment]
haha, recorded would work super great for me, but will be live answering questions, that will work better as I will be in control of the recording time and will answer any question on the fly during the broadcast

**Comment by radaniba** [comment]
what will be the dates (start - end) of this summer session ? 2 months ? 3 ?

**Comment by katieannemills** [comment]
@radaniba nominally May through August, but it's all pretty flexible - the determining factor is, as always, the whim and convenience of the speakers!

**Comment by minisciencegirl** [comment]
Looks like we have many great sessions being proposed here! Definitely keep them coming!

Next up: we need to decide on a good meeting time for everyone. As per usual, we plan on meeting up once a week, for about an hour. Please highlight your available times in this poll: http://whenisgood.net/zqr53wz 

Let's try to fill out the poll by May 6 so we can get the session started!

Note that the calendar day doesn't matter, just the day of the week for our usual thing. See you all soon!

**Comment by jennybc** [comment]
I'd also like to talk about [`purrr`](https://github.com/hadley/purrr), as I want to incorporate it into STAT 545 this fall. So @wmhall and I could maybe team up? Either create one lesson jointly or coordinate our stuff to be complementary? There is plenty to talk about.

**Comment by katieannemills** [comment]
@jennybc excellent! Will do. It's looking like we're going to be assembling a schedule the week of May 9, so we will post possible date then.

**Comment by wmhall** [comment]
@jennybc teaming up sounds great. It might make the most sense to just coordinate our stuff to be complementary. I was thinking of focusing on using nested data frames (via `tidyr`) in a split/apply/combine workflow, so `purrr` wouldn't be my focus. Maybe when scheduling happens we can try and make sure that my session comes soon after yours so that people are already comfortable with `purrr`

**Comment by SimonGoring** [comment]
What about "documentation", how to write good documentation, what works, what doesn't, how to host it, where to host it, &cetera.

**Comment by minisciencegirl** [comment]
This sounds awesome! @SimonGoring: would you be willing to run your Shiny/Leaflet session again? Sorry for missing it last time!

**Comment by morien** [comment]
@SimonGoring that's a great idea. documentation is key.

**Comment by radaniba** [comment]
👍  @SimonGoring  for documentation great idea, python users could be interested in [sphinx](http://www.sphinx-doc.org/en/stable/)

**Comment by jennybc** [comment]
@wmhall Agreed we can coordinate. `purrr` comes up in the workflow you discuss or, at least, it _can_. Because it can be really handy inside `mutate()` to work on the list columns that result from nesting more complicated structures in a data frame.

So yes I could give a more general `purrr` overview, which would then set it up for a mention by you. There's definitely plenty for two different-but-complementary sessions here.

**Comment by SimonGoring** [comment]
@radaniba @morien yes, I've been using sphinx and readthedocs.  I've figured out some fun stuff (like embedding code into the documentation), checked out a few templates & stuff.  I think I could do a general session on Sphinx & readthedocs (although I'm really a sphinx neophyte) and we could maybe have a general talk about good documentation, does anyone want to add some `roxygen2` talk to that? @jennybc or other active package developers? 

@minisciencegirl I can re-do the leaflet if there's interest.

**Comment by jennybc** [comment]
@SimonGoring I think some within rOpenSci (not including me) recently took a pretty hard look at readthedocs as we (this time including me) were interested in whether more tools and automation around shoving R documentation into that format and site would be worthwhile. I think the conclusion was equivocal at best. I'm not sure of all the details, but pretty sure there's no decision to go full steam ahead on that. So I would suggest steering our sessions towards roxygen2, vignette writing, READMEs, example coverage, etc. There is so much to talk about just with the immediate tools at hand.

**Comment by SimonGoring** [comment]
@jennybc absolutely.  I've just been working with it lately to do broader documentation for Neotoma (both [Tilia](http://tilia-manual.readthedocs.io/en/latest/) and [Neotoma](http://neotoma-manual.readthedocs.io/en/latest/) manuals are there) and have some experience so I offered when @radaniba mentioned it :)

I think of it more as a supplement than a central requirement.

**Comment by katieannemills** [comment]
Alright, the people have spoken - our preferred time for the summer is a three-way tie between Tuesdays at 10 AM, or Fridays at 1 or 4 PM. I nominate Fridays at 1 as a good way to wind down the week, but speak up if there are any strong opinions for the other two favorites.

In which case, we can start booking these awesome sessions you've suggested! What's say we start not this Friday but next? Also, there have been a bunch of requests for more introductory level lessons from new students; @JoeyBernhardt is going to help organize a run through of the [R Utility Belt](https://github.com/minisciencegirl/studyGroup/issues/29) you all helped outline last year - let us know if you're interested in teaching one of those!

All of which adds up to a summer schedule that looks like:

May 20 - _____________
May 27 - R Utility Belt: ggplot / @lmguzman 
June 3 - Coworking
June 10 - R Utility Belt
June 17 - _____________
June 24 - R Utility Belt
~~July 1 - Canada Day Holiday~~
July 8 - R Utility Belt
July 15 - _____________
July 22 - R Utility Belt
July 29 - Coworking
August 5 - R Utility Belt
August 12 - _____________
August 19 - R Utility Belt
August 26 - Coworking

A couple of points:
- The ______ are open spots for any of the suggestions above (or anything else anyone would like to propose; I think there are more ideas above than slots, so **feel free to sub in a topic suggestion instead of one of the Coworking sessions** to make more space. @wmhall @radaniba @markjohnsonubc @SimonGoring @jrebane @jennybc (and anyone else), please call out which day you want!
- The R Utility Belt can be taught in any order; feel free to call dibs on topics (listed at the top of [this issue](https://github.com/minisciencegirl/studyGroup/issues/29), and on Utility Belt days; I kept these really regular for the sake of the new students who only want to come to these intro sessions.

Any other comments or requests, post 'em here - I'm psyched for this lineup, it's going to be a fun summer!

**Comment by lmguzman** [comment]
I can do ggplot for the R utility belt, but I will only be here until June 10 :)

**Comment by katieannemills** [comment]
@lmguzman awesome! Would you like the May 27 slot for ggplot?

**Comment by lmguzman** [comment]
Sounds good

**Comment by SimonGoring** [comment]
How are the R Utility Belt modules saved?  I could help contribute to the basic stats one, building it in RMarkdown through a shared repo if we want.

Also I told @minisciencegirl I could re-run the shiny/leaflet module if requested.  I'm gone most of July, also for the Aug 12th and the June 17th :)  I guess I could do the May 20 meeting.  That seems soon since I just ran it last semester.

**Comment by radaniba** [comment]
@BillMills  Starting from July I can "broadcast" **Machine Learning with Python** you need just to specify the time that needs to be allocated for that (30 min? 1 hour ? more ? less ?). With it being online, you can feel free to spread the tutorials as you wish as I will be preparing a series of talks/demo online (youtube).

**Comment by radaniba** [comment]
btw we probably need another issue with only the plan, as this one will go on and on quickly with comments :) @BillMills you agree ?

**Comment by JoeyBernhardt** [comment]
Fridays at 1 sounds good to me!

**Comment by katieannemills** [comment]
@SimonGoring: [ggplot](https://gist.github.com/kazi11/22fac11fecd5dd5121b2), [dplyr + magrittr](https://github.com/ksamuk/dplyr_magrittr_demo) and [tidyr + reshape2](http://remi-daigle.github.io/tidyr_reshape2_lesson/) have been taught previously and have repos for the notes; I believe @eacton has notes for knitr + rmd, and the rest are still _dream stage_ - if you can help whip up multcomp + `lm()` + `glm()` (or related), that would be awesome!

If those dates don't work, what about a coworking day? We can definitely drop / split a coworking to accommodate!

**Comment by katieannemills** [comment]
@radaniba that is _awesome_, I'm super psyched for this! We should definitely pull in Study Groups worldwide if all they have to do is tune in. Will they be live broadcasts so there can be a question period? Sounds good either way!

I would recommend <=1h. bite-sized 30 minute chunks would be even better, but as this is a pretty advanced topic, it might be hard to get where you want to go in only 30 min - but if you can, go for it!
